runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -291,6 +291,14 @@ J9::CodeGenerator::fastpathAcmpHelper(TR::Node *node, TR::TreeTop *tt, const boo
             TR::Node* temp = expectedDeps->getChild(i);
             if (temp->getGlobalRegisterNumber() == depNode->getGlobalRegisterNumber())
                continue;
+            else if (temp->getOpCodeValue() == TR::PassThrough)
+               {
+               // PassThrough nodes cannot be commoned because doing so does not
+               // actually anchor the child, causing it's lifetime to not be extended
+               auto regnum = temp->getGlobalRegisterNumber();
+               temp = TR::Node::create(TR::PassThrough,1, temp->getFirstChild());
+               temp->setGlobalRegisterNumber(regnum);
```

Comment:
Rahil @r30shah drew my attention to some code in [JProfilingValue.cpp](https://github.com/eclipse/openj9/blob/a7bf6c09925da0a4e352114e63b3c2cb3f59b355/runtime/compiler/optimizer/JProfilingValue.cpp#L764-L781) that copies register dependencies.  It calls both `setLowGlobalRegisterNumber` and `setHighGlobalRegisterNumber` on the `PassThrough` node -  I guess to handle the possibility of register pairs.  Should this do that same?

Reply:
[setGlobalRegisterNumber](https://github.com/eclipse/omr/blob/0ec72973ff4a74a30dc1c38e8fc2e8e62de29134/compiler/il/OMRNode.cpp#L4938-L4944) sets the High register for the pass through node to -1 which can lead to issues if your node needs register pair.

Reply:
Fixed in force push.